### DISCUSSION TITLE
Delay in Obtaining WebRTC Stream

### DISCUSSION THREAD

**Discussion by SMDIndomitable** [bluenviron/mediamtx#3931] [Questions]
### Question

Hi, I need some urgent help if possible. I notice some delay in local WebRTC streaming without internet. I already consulted a similar thread raised by another person, but cant seems to figure out. I have ensured that the webrtcICEServers2: [] is empty, is there anything else I need to change?

**Comment by asturl** [comment]
We're gonna need more information about your "setup". What is the stream source?

**Comment by SMDIndomitable** [comment]
Hi, I am streaming from a USB Webcam (UVC) via rtsp to mediamtx, and grabbing the webrtc from mediamtx after. Sorry for the late reply! I was expecting the connection to be faster since is local, but it take around 17 seconds before I can see my webrtc stream

GSTPipelineStream = (
                        "appsrc ! "
                        "video/x-raw,format=BGR ! "
                        "queue ! "
                        "videoconvert ! "
                        "video/x-raw,format=BGRx ! "
                        "nvvidconv ! "
                        "nvv4l2h264enc bitrate=20000000 ! "
                        "video/x-h264,level=(string)4,profile=high ! "
                        "h264parse ! "
                        "rtspclientsink location=rtsp://localhost:8554/mystream"
                    )
This is what I am using to send over to mediamtx. For the webrtc is a React + Vite application

**Reply by asturl** [reply]
Does the stream start up faster when you directly display the restreamed RTSP source through mediamtx? As in, opening rtsp of "mystream" with a program that can display h264 rtsp streams directly (gstreamer itself or whatever)

Just trying to make sure the issue isn't in your encoding pipeline.
